### PR TITLE
simplify Windows development guide

### PR DESCRIPTION
**Pull request by friism**
This compresses quite a few of the steps and the user doesn't end up with a functioning Go environment on their host system. It's up for debate whether the simpler result is worth that trade-off.

Signed-off-by: Michael Friis friism@gmail.com

### CODE REVIEW

File: opensource/project/software-req-win.md [hunk 1]
```diff
@@ -34,229 +39,41 @@ fine. All the Docker project repositories are public and visible to everyone.
...
-12. Enter `GOPATH` for the **Variable Name**.
+Create a builder-container with the Docker source code. You can change the source code on your system and rebuild at your leisure:
 
-13. For the **Variable Value** enter the following:
+    docker  build -t docker-builder -f .\Dockerfile.windows .
 
-		C:\gopath;C:\gopath\src\github.com\docker\docker\vendor
+To build Docker, run:
 
+    docker run --name out docker-builder sh -c 'cd /c/go/src/github.com/docker/docker; hack/make.sh binary'
```
Comment (thaJeztah) [inline opensource/project/software-req-win.md]: Should we use a bind-mount (like we use with `BIND_DIR=.` on Linux?) that way people can make code changes on their host, and rebuild in the container, also it saves them from copying out the binary (as is explained in the next step)

File: opensource/project/software-req-win.md [hunk 2]
```diff
@@ -34,229 +39,41 @@ fine. All the Docker project repositories are public and visible to everyone.
...
 
-14. Press **OK** to close this dialog.
+You can copy out the resulting Windows Docker daemon binary:
 
-	The system adds `GOPATH` to the list of **System Variables**.
+    docker cp out:C:\go\src\github.com\docker\docker\bundles\$(cat VERSION)\binary-daemon\dockerd-$(cat VERSION).exe dockerd.exe
 
-15. Press **OK** twice to close out of the remaining dialogs.
+To test it, stop the system Docker daemon and start the one you just built:
```
Comment (friism) [inline opensource/project/software-req-win.md]: No.

**Comment by thaJeztah** [comment]
> and the user doesn't end up with a functioning Go environment on their host system. It's up for debate whether the simpler result is worth that trade-off.

I think it's a good trade-off, for Linux we also explain how to build in a build-container; https://docs.docker.com/opensource/project/set-up-dev-env/

Alternatively, we could keep the old document and make it an "alternative approach"

/cc @mstanleyjones

**Comment by thaJeztah** [comment]
Thanks @friism!

also /cc @johnstep to see if he has input / suggestions 😇

**Comment by friism** [comment]
@jhowardmsft might be interested too

**Comment by jhowardmsft** [comment]
Yeah, this definitely needs a refresh. I have a document I wrote about a year back which explains my full end to end native development environment. It's way out of date, but I should find time (haha!) to refresh it and publish it somehow. It covers the whole thing - how to setup and run CI, setup a dev VM, how to debug, build locally, in a container, shortcuts for dynamic binaries and so on.

**Comment by thaJeztah** [comment]
@jhowardmsft sounds great! We should probably split between the "express lane" (i.e. getting people ready for contributing in an easy way), and "advanced" setup

File: opensource/project/software-req-win.md (hunk 1 above)
Comment (friism) [inline opensource/project/software-req-win.md]: @thaJeztah according to John, that doesn't work

Comment (jhowardmsft) [inline opensource/project/software-req-win.md]: It doesn't. See my PR submitted this evening to docker-docker. Only on phone. Can't see PR# easily.

File: opensource/project/software-req-win.md (hunk 2 above)
Comment (jhowardmsft) [inline opensource/project/software-req-win.md]: Yup. You can't.

**Comment by friism** [comment]
@thaJeztah added a few tweaks and responded to comments. ptal

**Comment by friism** [comment]
@mstanleyjones addressed your comments. Starting the recently built daemon works because you stop the system one first (with `Stop-Service docker`).

**Comment by friism** [comment]
Ideally these instructions parallel the ones embedded in `Dockerfile.windows`. This is mostly the case, with this update: https://github.com/docker/docker/pull/27616/files

File: opensource/project/software-req-win.md [hunk 3]
```diff
@@ -9,254 +10,68 @@ menu:
...
-	In the next steps, you create environment variables for you Go paths.
+To contribute to the Docker project, you need a <a
+href="https://github.com" target="_blank">GitHub account</a>. A free account is
+fine. All the Docker project repositories are public and visible to everyone.
 
-5. Open the **Control Panel** on your system.
+This guide assumes that you have basic familiarity with Git and Github terminology and usage. Refer to [GitHub For Beginners: Don’t Get Scared, Get Started](http://readwrite.com/2013/09/30/understanding-github-a-journey-for-beginners-part-1/) to get up to speed on Github.
 
-6. Choose **System and Security > System**.
+## 3. Install Git for Windows
```
Comment (mstanleyjones) [inline opensource/project/software-req-win.md]: Do you think you should punctuate these sentences? I think I can see why you didn't, but I think I would prefer it.... It's not a deal breaker.

**Comment by friism** [comment]
@mstanleyjones I shortened the section titles to not be sentences - let me know what you think

File: opensource/project/software-req-win.md [hunk 4]
```diff
@@ -9,254 +10,68 @@ menu:
...
-
-	Compiling the source also adds a number of variables to your Windows environment.
-
-## Task 6. Get the Docker repository
-
-In this step, you start a Git `bash` terminal and get the Docker source code
-from GitHub.
-
-1. Locate the **Git Bash** program and start it.
+# Building and testing Docker on Windows
```
Comment (johndmulhausen) [inline opensource/project/software-req-win.md]: FYI, the title for docs is now set in the metadata, under `title:`

**Comment by johndmulhausen** [comment]
Deploy preview processing.

Built with commit ac139d7ae26337dfa781e2b636cf5fef4bdeff05

https://app.netlify.com/sites/docker-docs/deploys/581f4c7571e20a464fbefbeb

File: opensource/project/software-req-win.md (hunk 4 above)
Comment (friism) [inline opensource/project/software-req-win.md]: @johndmulhausen done

**Comment by thaJeztah** [comment]
@johndmulhausen looks like the netlify failures cannot be viewed publicly

**Comment by friism** [comment]
@mstanleyjones done

**Comment by mstanleyjones** [comment]
LGTM and Netlify worked!

**Comment by mstanleyjones** [comment]
@johndmulhausen if you are happy, please approve and merge.

**Comment by johndmulhausen** [comment]
@friism @thaJeztah @mstanleyjones Thanks, all :)

**Pull request merged**